### PR TITLE
feat: Add output transformation to IsDate

### PR DESCRIPTION
**Pull request by haakobja**
Added a serialization property to IsDate to allow for transforming Dates to string in correct format.
If `format='date'` the output is a date formatted string, on the format `yyyy-mm-dd`.
If `format='date-time'` the output is a ISO formatted date-time string

### CODE REVIEW

File: src/decorators/is-date.spec.ts [hunk 1]
```diff
@@ -29,12 +34,39 @@ describe('IsDate', () => {
...
+
+    it('transforms to date formatted string', async () => {
+      const dateToUse = new Date();
+      const obj = new TestSerialize();
+      obj.date = dateToUse;
+
+      expect(await output(obj)).toStrictEqual({ date: dateToUse.toISOString().slice(0, 10) });
+    });
+
+    it('pass through when field is undefined', async () => {
```
Comment (mdovhopo) [inline src/decorators/is-date.spec.ts:46]: I think this should throw, there is no 'optional' property in the parameters

File: src/decorators/is-date.spec.ts [hunk 2]
```diff
@@ -29,12 +34,39 @@ describe('IsDate', () => {
...
+      const obj = new TestSerialize();
+
+      expect(await output(obj)).toStrictEqual({});
+    });
+
+    it('pass through value if no serialize', async () => {
```
Comment (mdovhopo) [inline src/decorators/is-date.spec.ts:52]: if the 'format' is specified, i think class should not accept plain Date objects

File: src/decorators/is-date.spec.ts [hunk 3]
```diff
@@ -29,12 +34,39 @@ describe('IsDate', () => {
...
 
   describe('single date-time', () => {
     class Test {
       @IsDate({ format: 'date-time' })
-      date!: Date;
+      date?: Date;
+    }
+
+    class TestSerialize {
+      @IsDate({ format: 'date-time', serialize: true })
```
Comment (mdovhopo) [inline src/decorators/is-date.spec.ts:68]: pls add a test case without `serialize` prop, not sure it's clear how it works from test cases

**Review by mdovhopo — CHANGES_REQUESTED** [review]

File: src/decorators/is-date.spec.ts (hunk 3 above)
Comment (haakobja) [reply src/decorators/is-date.spec.ts:68]: Sure, I'll add some additional test cases. I also don't think `serialize` is the best name for a property, so I'll look into changing it

File: src/decorators/is-date.spec.ts (hunk 1 above)
Comment (haakobja) [reply src/decorators/is-date.spec.ts:46]: Yep, I agree that i should throw, not silently pass through the value

**Comment by glebbash** [comment]
This was a bug and it should not require a flag to work properly. Used your PR for reference to fix it in https://github.com/glebbash/nestjs-swagger-dto/commit/07e033a894d72a9f5a3e8519dde574acccc3df5e

**Pull request closed without merging**